nest-forms-backend/test/fixtures/auth/user-fixture-factory.ts

```diff
@@ -0,0 +1,321 @@
+import { UnauthorizedException } from '@nestjs/common'
+import { TestingModuleBuilder } from '@nestjs/testing'
+import {
+  randCompanyName,
+  randEmail,
+  randFirstName,
+  randLastName,
+} from '@ngneat/falso'
+import jwt from 'jsonwebtoken'
+import {
+  UserOfficialCorrespondenceChannelEnum,
+  UserVerifyStateCognitoTierEnum,
+  UserVerifyStateTypeEnum,
+} from 'openapi-clients/city-account'
+import { v4 as uuidv4 } from 'uuid'
+
+import { CityAccountUserService } from '../../../src/auth-v2/services/city-account-user.service'
+import { CognitoGuestIdentityService } from '../../../src/auth-v2/services/cognito-guest-identity.service'
+import { CognitoJwtVerifyService } from '../../../src/auth-v2/services/cognito-jwt-verify.service'
+import { CognitoUserService } from '../../../src/auth-v2/services/cognito-user.service'
+import { AuthUser, GuestUser, UserType } from '../../../src/auth-v2/types/user'
+
+export type GuestFixtureUser = {
+  identityId: string
+  headers: {
+    'X-Cognito-Guest-Identity-Id': string
+  }
+  user: GuestUser
+}
+export type AuthFixtureUser = {
+  sub: string
+  headers: {
+    Authorization: string
+  }
+  user: AuthUser
+}
+
+export type FixtureUser = GuestFixtureUser | AuthFixtureUser
+
+const getRandomGuestIdentityId = () => `eu-central-1:${uuidv4()}`
+
+export class UserFixtureFactory {
+  private readonly mockJwtSignSecret = 'secret'
+
+  private authUsers: AuthFixtureUser[] = []
+
+  private guestUsers: GuestFixtureUser[] = []
+
+  private generatedIcos = new Set<string>()
+
+  createGuestUser(): GuestFixtureUser {
+    const randomGuestIdentityId = getRandomGuestIdentityId()
+
+    const user: GuestFixtureUser = {
+      identityId: randomGuestIdentityId,
+      headers: {
+        'X-Cognito-Guest-Identity-Id': randomGuestIdentityId,
+      },
+      user: {
+        type: UserType.Guest,
+        cognitoIdentityId: randomGuestIdentityId,
+      },
+    }
+
+    this.guestUsers.push(user)
+    return user
+  }
+
+  private createAuthUser(
+    params:
+      | {
+          accountType: typeof UserVerifyStateTypeEnum.Fo
+          tier?: UserVerifyStateCognitoTierEnum
+        }
+      | {
+          accountType:
+            | typeof UserVerifyStateTypeEnum.FoP
+            | typeof UserVerifyStateTypeEnum.Po
+          ico: string
+          tier?: UserVerifyStateCognitoTierEnum
+        },
+  ): AuthFixtureUser {
+    const randomSub = uuidv4()
+    const nowInSeconds = Math.floor(Date.now() / 1000)
+    const cognitoPayloadRaw = {
+      sub: randomSub,
+      iss: 'https://cognito-idp.eu-central-1.amazonaws.com/eu-central-1_GCBQzfACy',
```

Comment:
Same as above, consider using env variable for this if there isn't reason not to and if feasible